### PR TITLE
Stream line and refactor of OAuth and External Call

### PR DESCRIPTION
**Pull request by siddharthashw**
Fixes - #192 

**1. IStringConstant.cs** - added GetAppDetailsFromClientAsyncUrl and update OAuthAfterLoginResponse name to OAuthAfterLoginResponseUrl and added new variables -> ExceptionMessageSlackUserNotFound and ExceptionMessageFailedToFetchDataException
**2. StringConstant.cs** - added GetAppDetailsFromClientAsyncUrl and update OAuthAfterLoginResponse name to OAuthAfterLoginResponseUrl. And update UserDetailsUrl, TeamLeaderDetailsUrl, ManagementDetailsUrl, ProjectUserUrl and CasualLeaveUrl and added new variables -> ExceptionMessageSlackUserNotFound and ExceptionMessageFailedToFetchDataException
**3. UserRepository.cs** - streamline
**4. IUserRepository.cs** - streamline
**5. ProjectUserController.cs** - deleted
**6. AutofacConfig.cs** - Deleted
**7. UserController.cs** - added projectusercontroller's API here
**8. BaseProvider.cs** - Appsetting configuration
**9. UserRepositoryTest.cs** - Added test cases for new External Call methods

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs [hunk 1]
```diff
@@ -271,558 +145,490 @@ public IEnumerable<UserAc> GetAllUsers()
                 user.NumberOfCasualLeave = editedUser.NumberOfCasualLeave;
                 user.NumberOfSickLeave = editedUser.NumberOfSickLeave;
                 user.SlackUserName = editedUser.SlackUserName;
-                var userPreviousInfo = await _userManager.FindByEmailAsync(editedUser.Email);
                 await _userManager.UpdateAsync(user);
+                //get user roles
                 IList<string> listofUserRole = await _userManager.GetRolesAsync(user);
-                var removeFromRole = await _userManager.RemoveFromRoleAsync(user, listofUserRole.FirstOrDefault());
+                //remove user role 
+                var removeFromRole = await _userManager.RemoveFromRoleAsync(user, listofUserRole.First());
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This is unused variable

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs [hunk 2]
```diff
@@ -271,558 +145,490 @@ public IEnumerable<UserAc> GetAllUsers()
...
+                //add new role of user.
                 var addNewRole = await _userManager.AddToRoleAsync(user, editedUser.RoleName);
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This is unused variable

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs [hunk 3]
```diff
@@ -271,558 +145,490 @@ public IEnumerable<UserAc> GetAllUsers()
...
-            IdentityResult result = await _userManager.ResetPasswordAsync(user, code, newPassword);
-            if (result.Succeeded)
-            {
-                _logger.LogInformation("Successfully Reset Password");
-                if (SendEmail(user, newPassword))
-                    return true;
-            }
-            return false;
+            var token = await _userManager.GeneratePasswordResetTokenAsync(user); //genrate passsword reset token
+            IdentityResult result = await _userManager.ResetPasswordAsync(user, token, newPassword);
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This is also unused variable

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs (hunk 1 above)
Comment (siddharthashw) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This method belong to someone else but still variable was not in use. I have removed it

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs (hunk 2 above)
Comment (siddharthashw) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This method belong to someone else but still variable was not in use. I have removed it

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs (hunk 3 above)
Comment (siddharthashw) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: This method belong to someone else but still variable was not in use. I have removed it

**Pull request merged**